app/assets/javascripts/ui_models/app_state/preferences_state.ts

```diff
@@ -0,0 +1,26 @@
+import { action, computed, makeObservable, observable } from 'mobx';
+
+export class PreferencesState {
+  private _open = true;
+
+  constructor() {
+    makeObservable<PreferencesState, '_open'>(this, {
+      _open: observable,
+      open: action,
+      close: action,
+      isOpen: computed,
+    });
+  }
+
+  open = (): void => {
```

Comment:
We may want to add a setter for this.open like this

setOpen(open: boolean): void {
  this.open = open;
}

and then we can either call that setter directly with true/false, or if you prefer to add two separate functions, maybe we could call them openPreferences or closePreferences to avoid confusion with the "open" attribute. Also not really important, but I think for state functions we're currently using named functions

Reply:
Done.